components/camel-bean/src/main/java/org/apache/camel/component/bean/BeanInfo.java

```diff
@@ -63,7 +63,9 @@ public class BeanInfo {
     private static final Logger LOG = LoggerFactory.getLogger(BeanInfo.class);
     private static final String CGLIB_CLASS_SEPARATOR = "$$";
     private static final String CGLIB_METHOD_MARKER = "CGLIB$";
+    private static final String BYTE_BUDDY_CLASS_SEPARATOR = "$ByteBuddy$";
     private static final String BYTE_BUDDY_METHOD_MARKER = "$accessor$";
+    private static final String ARIES_PROXY_CLASS_PREFIX = "Proxy";
```

Comment:
This is not accepted as Proxy has no special marker that would cause problems for end users that have normal classes that are named `ProxyMyWorld` or something.

Why does Aries not use a special marker like the others do

Reply:
The class name of aries-proxy class is created as:
```
className = "Proxy" + AbstractWovenProxyAdapter.getSanitizedUUIDString();
```
So it would be possible to create a regex for it, if we wanted to be more precise. But the class name is created in default java package, so it should be pretty resistant to real world user classes. At least that was how I thought about it.

Reply:
Or the check could be done via reflection, checking for some other markers.

Reply:
Reflection should be avoided.